nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java

```diff
@@ -3150,4 +3150,22 @@ public void springAutowiredConstructorTest() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void derefNullableTernary() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "public class Test {",
+            "  public void derefTernary(boolean b) {",
+            "    Object o1 = null, o2 = new Object();",
+            "    // BUG: Diagnostic contains: dereferenced expression (b ? o1 : o2) is @Nullable",
+            "    (b ? o1 : o2).toString();",
```

Comment:
Can we test `(b ? o2 : o1).toString()`? Both for completeness and so I don't confuse it with the safe case below because of reading in a hurry 😅

Reply:
5082288